### ISSUE TITLE
QCPortal: ReactionDataset.query_components function

### ISSUE THREAD

**Issue by mattwelborn** [MolSSI/QCFractal#378]
**Is your feature request related to a problem? Please describe.**
It is unwieldy to get data from a ReactionDataset which is not a products-minus-reactants observable.

**Describe the solution you'd like**
A `query_components` function analogous to `query` which returns a DataFrame of fields (e.g. energy, Result, stdout) for reactants and products.

Example output:
```
# Methane Dimer | cp | 0 | / coefficient / field
# Methane Diemr / nocp / 0 / coefficient / field
# Methane Diemr / nocp / 1 / coefficient/ field
```

**Describe alternatives you've considered**
This can be currently achieved with:
```
mol_id = ds.data.records[0].stoichiometry["default"]
ret = client.query_results(molecule=5, method="B3lYP", basis="aug-cc-pVDZ")
print(ret[0].get_stdout())
```

**Comment by mattwelborn** [comment]
@dgasmith please @ me on the PR so I can see how this works.

**Comment by mattwelborn** [comment]
Resolved by PR #385. This functionality is in get_molecules and get_records.
